admin/app/Http/Controllers/PageController.php

```diff
@@ -116,6 +117,13 @@ public function store(Request $request)
      */
     public function destroy(Request $request, Page $page)
     {
+        
+        // delete corresponding link in menu
+        $pageMenuItem = MenuItem::where('link', 'route://site.'.$page->id)->first();
+        if ($pageMenuItem) {
+            $pageMenuItem->update(['link' => null]);
+        }
```

Comment:
Actually, this fix doesn't consider cases where the page is linked to more than menu item (why ffs one should do that).
```suggestion
        MenuItem::where('link', 'route://site.'.$page->id)->update(['link' => null]);
```
This should work as well and would update all items where this page was linked.